src/DistributedMetadata/CatalogService.cpp

```diff
@@ -411,6 +411,35 @@ bool CatalogService::tableExists(const String & database, const String & table)
     return indexed_by_name.find(std::make_pair(database, table)) != indexed_by_name.end();
 }
 
+std::pair<bool, bool> CatalogService::columnExists(const String & database, const String & table, const String & column) const
+{
+    String create_table_query;
+    {
+        std::shared_lock guard{catalog_rwlock};
+        auto iter = indexed_by_name.find(std::make_pair(database, table));
+        if (iter == indexed_by_name.end())
+        {
+            return {false, false};
+        }
+        create_table_query = iter->second.begin()->second->create_table_query;
```

Comment:
nit: Another way is use a get a `TablePtr` and then we don't need a copy of the create table query statement. Since TablePtr is shared_ptr, we don't need worry about somebody destruct it or it gets changes without hold the lock (we don't update a table inplace right now).

Essentially we like to protect `indexed_by_name` data structure by using lock, but not the found result

Reply:
fixed